eox_nelp/pearson_vue/pipeline.py

```diff
@@ -392,18 +414,98 @@ def build_ead_request(
     Returns:
         dict: dict with ead_request dict
     """
-    ead_request = {
-        "@clientAuthorizationID": exam_metadata["client_authorization_id"],
-        "@clientID": getattr(settings, "PEARSON_RTI_WSDL_CLIENT_ID"),
-        "@authorizationTransactionType": transaction_type,
-        "clientCandidateID": f'NELC{profile_metadata["anonymous_user_id"]}',
-        "examAuthorizationCount": exam_metadata["exam_authorization_count"],
-        "examSeriesCode": exam_metadata["exam_series_code"],
-        "eligibilityApptDateFirst": exam_metadata["eligibility_appt_date_first"],
-        "eligibilityApptDateLast": exam_metadata["eligibility_appt_date_last"],
-        "lastUpdate": timezone.now().strftime("%Y/%m/%d %H:%M:%S GMT"),
-    }
+    try:
+        ead_request = {
+            "@clientAuthorizationID": exam_metadata["client_authorization_id"],
+            "@clientID": getattr(settings, "PEARSON_RTI_WSDL_CLIENT_ID"),
+            "@authorizationTransactionType": transaction_type,
+            "clientCandidateID": f'NELC{profile_metadata["anonymous_user_id"]}',
+            "examAuthorizationCount": exam_metadata["exam_authorization_count"],
+            "examSeriesCode": exam_metadata["exam_series_code"],
+            "eligibilityApptDateFirst": exam_metadata["eligibility_appt_date_first"],
+            "eligibilityApptDateLast": exam_metadata["eligibility_appt_date_last"],
+            "lastUpdate": timezone.now().strftime("%Y/%m/%d %H:%M:%S GMT"),
+        }
+    except KeyError as exc:
+        raise PearsonKeyError("EAD", str(exc)) from exc
+    except AttributeError as a_exc:
+        raise PearsonAttributeError("EAD", str(a_exc)) from a_exc
 
     return {
         "ead_request": ead_request
     }
+
+
+def audit_pearson_error(exception_data=None, hidden_kwargs=None, **kwargs):
+    """
+    Method to save an error with eox-audit.
+    Args:
+        exception_data(dict): dict with exception_type key.
+        hidden_kwargs(list: str): List with keys name that dont want to be added
+        the audit model and the logger error.
+        **kwargs
+    Logs:
+        LogError: Log everything with name error.
+    Returns:
+        None
+    """
+    if not exception_data:
+        return
+    if hidden_kwargs is None:
+        hidden_kwargs = []
+    hidden_kwargs.append("hidden_kwargs")  # Clean also the same hidden.
+    audit_kwargs = remove_keys_from_dict(kwargs, hidden_kwargs)
+
+    audit_action = "Pearson Vue Exception"
+    audit_action = f"{audit_action}~{exception_data['exception_type']}"
+    pearson_exception = find_class_with_attribute_value(
+        exceptions,
+        "exception_type",
+        exception_data['exception_type'],
+    ) or PearsonBaseError
+
+    @audit_method(action=audit_action)
+    def raise_audit_pearson_exception(exception_data, **audit_kwargs):
+        raise pearson_exception(exception_data, audit_kwargs)
```

Comment:
Here you are passing two positional arguments so   if the pearson_exception is PearsonAttributeError that equivalent to `pearson_exception(request_type=exception_data, attribute_error=audit_kwargs)`, does that make sense ?
id f the exception_data contains  all the original data that should be `pearson_exception(**exception_data)`,